mvpa2/tests/test_hyperalignment.py

```diff
@@ -315,6 +319,52 @@ def _test_on_swaroop_data(self):  # pragma: no cover
         pass
 
 
+def test_searchlight_hyperalignment():
+    # toy data
+    data = np.random.randn(18,4,2)
+    space = 'voxel_indices'
+    # total number of datasets for the analysis
+    nds = 5
+    ds_orig = Dataset.from_wizard(data, space=space)
+    dss = [ds_orig]
+    # create  a few distorted datasets to match the desired number of datasets
+    dss += [local_random_affine_transformations(
+                ds_orig,
+                scatter_neighborhoods(Sphere(8),
+                                      ds_orig.fa[space].value,
+                                      deterministic=False)[1],
+                Sphere(1),
+                space=space)
+                    for i in range(nds-1)]
+    # we should have some distortion
+    for ds in dss[1:]:
+        assert_false(np.all(ds_orig.samples == ds.samples))
+
+    # store projections for each mapper separately
+    projs = [list() for i in range(nds)]
+    # run the algorithm with all combinations of the two major parameters
+    # for projection calculation.
+    for kwargs in [{'combine_neighbormappers': True, 'sparse_mode': 'coo'},
+                   {'combine_neighbormappers': True, 'dtype': 'float64'},
+                   {'combine_neighbormappers': False, 'sparse_mode': 'coo'},
+                   {'combine_neighbormappers': False, 'compute_recon': 'False'},
+                   {'combine_neighbormappers': True, 'exclude_from_model': [2,4]}]:
+        slhyp = SearchlightHyperalignment(**kwargs)
+        print 'nds:', len(dss)
+        mappers = slhyp(dss)
+        # one mapper per input ds
+        assert_equal(len(mappers), nds)
+        for midx, m in enumerate(mappers):
+            projs[midx].append(m.proj)
+
+    for midx, m in enumerate(mappers):
+        # different sparse modes yield the same projections
+        # without combine_neighbormappers
+        assert_array_almost_equal(projs[midx][2].todense(), projs[midx][3].todense())
+        # with combine_neighbormappers
+        assert_array_almost_equal(projs[midx][0].todense(), projs[midx][1].todense())
```

Comment:
so this test only tests that they produce the same result, but not necessarily anywhere near correct one.  How would we know that it does any good?